src/energyplus/ForwardTranslator/ForwardTranslateZoneHVACEvaporativeCoolerUnit.cpp

```diff
@@ -0,0 +1,247 @@
+      outdoorAirInletNodeName = node->name().get();
+      idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::OutdoorAirInletNodeName, node->name().get());
+    }
+
+    // Cooler Outlet Node Name: Required Node
+    if (boost::optional<Node> node = modelObject.outletNode()) {
+      coolerOutletNodeName = node->name().get();
+      idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::CoolerOutletNodeName, node->name().get());
+    }
+
+    // Supply Air Fan Object Type
+    // Supply Air Fan Name
+    boost::optional<IdfObject> fan_;
+    if (boost::optional<HVACComponent> supplyAirFan = modelObject.supplyAirFan()) {
+      fan_ = translateAndMapModelObject(supplyAirFan.get());
+
+      if (fan_ && fan_->name()) {
+        idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::SupplyAirFanObjectType, fan_->iddObject().name());
+        idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::SupplyAirFanName, fan_->name().get());
+      }
+    }
+
+    if (modelObject.isDesignSupplyAirFlowRateAutosized()) {
+      idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::DesignSupplyAirFlowRate, "Autosize");
+    } else {
+      // Design Supply Air Flow Rate: boost::optional<double>
+      if (boost::optional<double> designSupplyAirFlowRate_ = modelObject.designSupplyAirFlowRate()) {
+        idfObject.setDouble(ZoneHVAC_EvaporativeCoolerUnitFields::DesignSupplyAirFlowRate, designSupplyAirFlowRate_.get());
+      }
+    }
+
+    // Fan Placement: Required String
+    const std::string fanPlacement = modelObject.fanPlacement();
+    idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::FanPlacement, fanPlacement);
+
+    // Cooler Unit Control Method: Required String
+    const std::string coolerUnitControlMethod = modelObject.coolerUnitControlMethod();
+    idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::CoolerUnitControlMethod, coolerUnitControlMethod);
+
+    // Throttling Range Temperature Difference: Optional Double
+    const double throttlingRangeTemperatureDifference = modelObject.throttlingRangeTemperatureDifference();
+    idfObject.setDouble(ZoneHVAC_EvaporativeCoolerUnitFields::ThrottlingRangeTemperatureDifference, throttlingRangeTemperatureDifference);
+
+    // Cooling Load Control Threshold Heat Transfer Rate: Optional Double
+    const double coolingLoadControlThresholdHeatTransferRate = modelObject.coolingLoadControlThresholdHeatTransferRate();
+    idfObject.setDouble(ZoneHVAC_EvaporativeCoolerUnitFields::CoolingLoadControlThresholdHeatTransferRate,
+                        coolingLoadControlThresholdHeatTransferRate);
+
+    // First Evaporative Cooler Object Type: Required String
+    // First Evaporative Cooler Object Name: Required Object
+    HVACComponent firstEvaporativeCooler = modelObject.firstEvaporativeCooler();
+    boost::optional<IdfObject> firstEvaporativeCooler_ = translateAndMapModelObject(firstEvaporativeCooler);
+    if (firstEvaporativeCooler_) {
```

Comment:
Assign in the if. 
Also, should we LOG_AND_THROW if /somehow/ the translateAndMapModelObject failed?